### PR TITLE
Add ngrok instructions

### LINKED ISSUE #85: Add instructions for Ngrok usage
For development and for local self-hosting, we're defaulting to using Ngrok for HTTPS URLs (needed for basically any external data source).

We need to add instructions on how to set up an Ngrok account with a static domain and how to set the key + domain in .env.

Relevant discussion: #66

### PR DESCRIPTION
**Pull request by Neumand**
/fix #85 

This PR adds instructions for setting up ngrok with a free static domain.

### CODE REVIEW

File: docker-compose.yml
Comment (Shpigford) [inline docker-compose.yml:1]: What happens if you *don't* setup ngrok and then try to run docker?

Comment (Neumand) [reply docker-compose.yml:1]: Good point.

It will set it up with an empty string. I'll change it to use `-host.docker.internal` as the default value.

**Review by jbolda — APPROVED** [review]

**Comment by Shpigford** [comment]
I'm currently getting the following error when trying to do this...

```
Your account is on the 'Pay-as-you-go' plan, which requires you to register a 
Domain through the ngrok Dashboard before starting an Endpoint. 
Visit https://dashboard.ngrok.com/cloud-edge/domains to register a domain 
and then use the `--domain` flag when starting your Agent.
```

I've registered a domain and have `NGROK_DOMAIN` set in `.env.`

**Comment by Neumand** [comment]
> I'm currently getting the following error when trying to do this...
> 
> ```
> Your account is on the 'Pay-as-you-go' plan, which requires you to register a 
> Domain through the ngrok Dashboard before starting an Endpoint. 
> Visit https://dashboard.ngrok.com/cloud-edge/domains to register a domain 
> and then use the `--domain` flag when starting your Agent.
> ```
> 
> I've registered a domain and have `NGROK_DOMAIN` set in `.env.`

Hmm, I assume you also have `NGROK_AUTH_TOKEN` set?

And if you visit https://dashboard.ngrok.com/cloud-edge/domains you can see your domain, right?

image

Strange. I'd like some others to try and see what happens. I'm not getting that error.

**Comment by Neumand** [comment]
@jbolda Can you try and see what happens?

**Comment by Shpigford** [comment]
Here's what i've got...

CleanShot 2024-01-18 at 11 16 45@2x

CleanShot 2024-01-18 at 11 17 35@2x

**Comment by Neumand** [comment]
Seems like it has to do with the account type.

It should work if you try with a different free account. Not sure how to proceed though...

image

**Pull request merged**